### PR TITLE
Server side image rendering

### PR DESCRIPTION
**Pull request by itamarst**
Based on #487.

* [x] Works in PySpark kernel.
* [x] Works in normal notebook.
* [x] Unit tests.
* [ ] Real-world testing by someone else.

### CODE REVIEW

**Comment by itamarst** [comment]
Testing is a bit iffy, but honestly only real-world use tests this actually works, and there's no good framework for end-to-end tests at the moment.

File: sparkmagic/sparkmagic/livyclientlib/command.py [hunk 1]
```diff
@@ -65,7 +68,14 @@ def _get_statement_output(self, session, statement_id):
                     return (True, u"")
 
                 if statement_output[u"status"] == u"ok":
-                    return (True, statement_output[u"data"][u"text/plain"])
+                    data = statement_output[u"data"]
+                    png_encoded = data.get("image/png")
+                    if png_encoded:
+                        image = Image(base64.b64decode(png_encoded))
+                        return (True, image)
+                    else:
```
Comment (Tagar) [inline sparkmagic/sparkmagic/livyclientlib/command.py]: IPython.display.Image should also support png/gif?

Comment (itamarst) [reply sparkmagic/sparkmagic/livyclientlib/command.py]: Yeah, probably, but Livy specifically only returns PNG at the moment.

**Comment by itamarst** [comment]
@Tagar does it work for you? There's examples in the notebooks.

**Comment by Tagar** [comment]
@itamarst I am wrapping up my work projects as I am preparing to go for an extended OOO
so don't have capacity to test this unfortunately. 

It would be great if you could merge this since you already did some testing.
More folks can test this as part of a release and 
we can submit fixes for any outstanding issues as necessary later?

Thank you for all those improvements you're submitting. 
Glad to see many of those long-standing spark-magic issues are finally getting resolved.

**Comment by itamarst** [comment]
I expect to be doing more work on Sparkmagic Thursday or Friday, which will include a new release with this and #541.

**Comment by Tagar** [comment]
Both are awesome improvements. Thanks a lot

**Comment by itamarst** [comment]
Thank you @qiaoshi.wang for the original patch!

**Pull request merged**